### PR TITLE
Cleanup some code

### PR DESCRIPTION
**Pull request by Flamefire**
- Reduce include directories so we have full path names, e.g. `#include libutil/src/Log.h" which helps to see where the classes come from
- `unsigned int` -> `unsigned` for brevity
- build_version only in main project

### CODE REVIEW

**Comment by Flow86** [comment]
please use uint32_t instead, so its clear "how big" the unsigned actually should be (or use size_t)

File: driver/CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ADD_SUBDIRECTORY( src )
 
 #################################################################################
-
+INCLUDE_DIRECTORIES(${PROJECT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/src)
```
Comment (Flow86) [inline driver/CMakeLists.txt]: his should not be necessary if you set the include paths correctly?

Comment (Flamefire) [inline driver/CMakeLists.txt]: PROJECT_SOURCE_DIR is for getting includes like `libendian/src/*` and the other one for the base driver files like the interface

**Comment by Flamefire** [comment]
You mean because of the `unsigned int`->`unsigned`? Did not want to introduce fixed width integers yet, just clean some stuff up and the replacement is equivalent, just shorter

**Comment by stefson** [comment]
I checked out your fork and the branch and applied the two recent commits from master on top of it. Turns out something is wrong with the changes in libsiedler2, isn't it? 


```[ 24%] Building CXX object libsiedler2/src/CMakeFiles/siedler2.dir/WriteBBM.cpp.o
cd /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999_build/libsiedler2/src && /usr/bin/x86_64-pc-linux-gnu-g++  -DBOOST_PP_VARIADICS=1 -DHAVE_MEMCHECK_H -DNOMINMAX -Doverride="" -I/var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999_build -I/var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999   -DNDEBUG -march=core2 -O2 -ggdb -ffast-math -mmmx -msse -mfpmath=sse -Wall -Werror -pedantic -Wpedantic -Wparentheses -Wno-error=type-limits -Wfloat-conversion -Wno-long-long -Wno-unknown-pragmas -fno-strict-aliasing -Wno-variadic-macros   -o CMakeFiles/siedler2.dir/WriteBBM.cpp.o -c /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/libsiedler2/src/WriteBBM.cpp
In file included from /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/libendian/src/EndianOStreamAdapter.h:23:0,
                 from /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/libsiedler2/src/WriteBBM.cpp:23:
/var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/libendian/src/EndianStreamAdapterBase.h: In instantiation of ‘libendian::EndianStreamAdapterBase<T_isBigEndian, T_Stream>::EndianStreamAdapterBase(const T_InitType&, const T_ArgType&) [with T_InitType = std::__cxx11::basic_string<char>; T_ArgType = std::_Ios_Openmode; bool T_isBigEndian = false; T_Stream = std::basic_ofstream<char>]’:
/var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/libendian/src/EndianOStreamAdapter.h:50:106:   required from ‘libendian::EndianOStreamAdapter<T_isBigEndian, T_Stream>::EndianOStreamAdapter(const T_InitType&, const T_ArgType&) [with T_InitType = std::__cxx11::basic_string<char>; T_ArgType = std::_Ios_Openmode; bool T_isBigEndian = false; T_Stream = std::basic_ofstream<char>]’
/var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/libsiedler2/src/WriteBBM.cpp:52:90:   required from here
/var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/libendian/src/EndianStreamAdapterBase.h:54:112: error: no matching function for call to ‘std::basic_ofstream<char>::basic_ofstream(const std::__cxx11::basic_string<char>&, const std::_Ios_Openmode&)’
         explicit EndianStreamAdapterBase(const T_InitType& initArg, const T_ArgType& arg): stream_(initArg, arg) {}
                                                                                                                ^
In file included from /var/tmp/portage/games-strategy/s25rttr-9999-r2/work/s25rttr-9999/libsiedler2/src/WriteBBM.cpp:22:0:
/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include/g++-v5/fstream:697:7: note: candidate: std::basic_ofstream<_CharT, _Traits>::basic_ofstream(const char*, std::ios_base::openmode) [with _CharT = char; _Traits = std::char_traits<char>; std::ios_base::openmode = std::_Ios_Openmode]
       basic_ofstream(const char* __s,
       ^
/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include/g++-v5/fstream:697:7: note:   no known conversion for argument 1 from ‘const std::__cxx11::basic_string<char>’ to ‘const char*’
/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include/g++-v5/fstream:682:7: note: candidate: std::basic_ofstream<_CharT, _Traits>::basic_ofstream() [with _CharT = char; _Traits = std::char_traits<char>]
       basic_ofstream(): __ostream_type(), _M_filebuf()
       ^
/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include/g++-v5/fstream:682:7: note:   candidate expects 0 arguments, 2 provided
/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include/g++-v5/fstream:656:11: note: candidate: std::basic_ofstream<char>::basic_ofstream(const std::basic_ofstream<char>&)
     class basic_ofstream : public basic_ostream<_CharT,_Traits>
           ^
/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include/g++-v5/fstream:656:11: note:   candidate expects 1 argument, 2 provided
```

**Comment by Flamefire** [comment]
Yeah C++11 feature. I'm working on libsiedler2 atm with some major changes which I want to finish first.

**Pull request merged**